src/compiler/checker.ts

```diff
@@ -19789,6 +19789,15 @@ namespace ts {
             }
         }
 
+        function checkJSDocParameterTag(node: JSDocParameterTag) {
+            checkSourceElement(node.typeExpression);
+            if (!getParameterSymbolFromJSDoc(node)) {

```

Comment:
`getParameterSymbolFromJSDoc` has a linear search of `function.parameters`, and we call this for every `@param` node. So it's quadratic where it could be linear if we wrote sufficiently clever code.

Might not be an issue, but I wanted to make sure you'd thought about it. (Most functions have few parameters and most `@params` are correct, so pathological cases should be rare.)